spec/libs/installations/crashlytics/api_spec.rb

```diff
@@ -0,0 +1,189 @@
+require "rails_helper"
+
+Dataset = Struct.new(:dataset_id, :project_id)
+APP_IDENTIFIER = "com.example.app"
+APP_ONE = "App One"
+APP_TWO = "App Two"
+
+describe Installations::Crashlytics::Api, type: :integration do
+  let(:project_number) { Faker::Number.number(digits: 8).to_s }
+  let(:json_key) { StringIO.new({client_email: "client@test.com", private_key: "private_key"}.to_json) }
+  let(:api_instance) { described_class.new(project_number, json_key) }
+
+  let(:app_id) { APP_IDENTIFIER }
+  let(:version) { "1.0.0" }
+  let(:build_number) { "100" }
+  let(:payload) do
+    {
+      new_errors_count: 5,
+      errors_count: 10,
+      sessions_in_last_day: 100,
+      sessions: 200,
+      sessions_with_errors: 50,
+      daily_users_with_errors: 30,
+      daily_users: 150,
+      total_sessions_in_last_day: 150,
+      external_release_id: "release_123"
+    }
+  end
+  let(:transforms) { CrashlyticsIntegration::RELEASE_TRANSFORMATIONS }
+
+  describe "#find_release" do
```

Comment:
**issue:** I think this test is of not much value. It just tests that the function calls another function that has been mocked.

The best version of this test would be a mock BQ with fixtures. But a slightly easier version would probably be to at least mock just the `query` function on the bq client.

Thumb-rule: we should only mock external library code / bg worker code / api calls being made externally – mocking your own code is a smell